app/grants/models.py

```diff
@@ -474,22 +469,24 @@ def get_hash_from_web3(self):
             args['nonce'],
             ).call()
 
-
-    def successful_contribution(self, kwargs):
+    def successful_contribution(self, tx_id):
         """Create a contribution object."""
         from marketing.mails import successful_contribution
         self.last_contribution_date = timezone.now()
-        self.next_contribution_date = timezone.now() + timezone.timedelta(seconds=self.real_period_seconds)
+        self.next_contribution_date = timezone.now() + timezone.timedelta(seconds=int(self.real_period_seconds))
         self.save()
+        # TODO: add gasPrice and nonce
         contribution_kwargs = {
-            'tx_id': kwargs.tx_id,
-            'gas_price': kwargs.gas_price,
-            'nonce': kwargs.nonce,
+            'tx_id': tx_id,
             'subscription': self
         }
         contribution = Contribution.objects.create(**contribution_kwargs)
         grant = self.grant
-        grant.amount_received = (grant.amount_received + convert_amount(self.amount_per_period, self.token_symbol, "USDT", timezone.now()))
+        grant.amount_received = (
```

Comment:
what if the token_symbol can't be converted to USDT?  seems to me we should handle this case..

Reply:
ok but we can't merge this PR into the stable branch if theres a `ConversionNotFound` exception that's not handled

Reply:
Gotcha. This PR is for `grantz-cleanup`. Not trying to move it `master`.  Wanting to get the functional miner where other can start using it though.

Reply:
its only a couple line fix to wrap this code in this PR in a try/catch, but if you want to create a github issue to track this and do it later, that works for me.